lib/ui/painting.dart

```diff
@@ -13,19 +13,19 @@ part of dart.ui;
 // would cause a crash. It is, however, acceptable for error messages to be much
 // less useful or correct in release mode than in debug mode.
 //
-// Painting APIs will also warn about arguments representing NaN or infinite
-// coordinates, which can not be rendered by Skia.
+// Painting APIs will also warn about arguments representing NaN coordinates,
+// which can not be rendered by Skia.
 
 bool _rectIsValid(Rect rect) {
-  return rect != null && rect.isFinite;
+  return rect != null && !rect.left.isNaN && !rect.top.isNaN && !rect.right.isNaN && !rect.bottom.isNaN;
 }
 
 bool _rrectIsValid(RRect rrect) {
-  return rrect != null && rrect.isFinite;
+  return rrect != null && !rrect.left.isNaN && !rrect.top.isNaN && !rrect.right.isNaN && !rrect.bottom.isNaN;
```

Comment:
We only call this is asserts, right?
If we only call this in asserts, it might be better to just have these methods always return true, but have them themselves contain two asserts, so we could give messages:

```dart
bool _debugRectIsValid(Rect rect) {
  assert(rect != null, 'Rect argument was null.');
  assert(!rect._value.any((double value) => value.isNaN), 'Rect argument contained a NaN value.');
  return true;
}
```